src/Address.svelte

```diff
@@ -2,37 +2,33 @@
   import { onMount } from 'svelte';
   import { link } from 'svelte-routing';
   import { ethers } from 'ethers';
-  import { safeLink, explorerLink, identifyAddress, formatAddress, AddressType } from '@app/utils';
+  import { safeLink, explorerLink, identifyAddress, formatAddress, AddressType, parseEnsLabel } from '@app/utils';
+  import { Profile } from '@app/profile';
   import Loading from '@app/Loading.svelte';
   import Avatar from "@app/Avatar.svelte";
   import type { Config } from '@app/config';
-  import type { Registration } from '@app/base/registrations/registrar';
-  import { getRegistration } from '@app/base/registrations/registrar';
 
   export let address: string;
   export let config: Config;
   export let resolve = false;
   export let noBadge = false;
   export let noAvatar = false;
   export let compact = false;
+  // This property allows components eg. Header.svelte to pass a resolved profile object.
+  export let profile: Profile | null = null;
```

Comment:
I actually like this change generally!

Reply:
Yeah I think it could be used to avoid building a user profile when a component already did all the heavy lifting.